tests/components/vesync/test_select.py

```diff
@@ -0,0 +1,43 @@
+"""Tests for the select platform."""
+
+from unittest.mock import patch
+
+from homeassistant.components.select import (
+    ATTR_OPTION,
+    DOMAIN as SELECT_DOMAIN,
+    SERVICE_SELECT_OPTION,
+)
+from homeassistant.components.vesync.const import NIGHT_LIGHT_LEVEL_DIM
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+
+from .common import ENTITY_HUMIDIFIER_300S_NIGHT_LIGHT_SELECT
+
+
+async def test_set_nightlight_level(
```

Comment:
```suggestion
async def test_night_light_level(
```

can we also test reading the current value?

Reply:
The device in the test is a mocked device so it itself would not update. The `select` entity update could however be a part of test_platform but it currently does not handle humidifier device at all.

Reply:
What I mean is that you can use the mocked device to set a state and ensure that the select state reflects it. The other tests for vesync do that (check `test_mist_level` in `test_number.py`, for example)